### ISSUE TITLE
Geometry Nodes - add icons, add new nodes to the sidebar and remove legacy nodes from the sidebar

### ISSUE THREAD

**Issue by ReinerBforartists** [Bforartists/Bforartists#2690]
Big change in the geometry nodes. The new node system is kicking in. The old nodes are hidden, the new nodes pops up.

The old nodes remains under the hood for compatibility reasons. So not sure what to do with it regarding the manual. I guess we need to split up the chapters.

Task: lots to do ... . Add icons, add new nodes to the sidebar. And remove the legacy nodes from the sidebar.

![addicons](https://user-images.githubusercontent.com/13919702/135751071-99522fd9-5517-4962-9619-b920c2ecad5a.jpg)

**Comment by Draise14** [comment]
This is big but still not ready for production. The legacy system is still included but as "experimental".

**Comment by ReinerBforartists** [comment]
Yes. But we need to stay close to the Blender changes. As told in the other thread, the missing changes will most probably arrive in the next weeks.

**Comment by ReinerBforartists** [comment]
Float Curve node is also changed in the shader editor. But has most probably the same code. And so the same icon. We will see.

![shader](https://user-images.githubusercontent.com/13919702/135886396-816238fc-8337-4fe2-9935-dc36f638ce2e.jpg)

**Comment by ReinerBforartists** [comment]
Let's do this task menu by menu.

**Comment by ReinerBforartists** [comment]
Attribute nodes

![attr](https://user-images.githubusercontent.com/13919702/135996401-c34c6fc0-a9a4-48ad-ab22-76042b1555fe.jpg)

**Comment by ReinerBforartists** [comment]
Curve nodes

![curve](https://user-images.githubusercontent.com/13919702/136073685-440f0c31-6c0b-4444-b32b-b937797a4b0f.png)

**Comment by ReinerBforartists** [comment]
Material nodes

![material](https://user-images.githubusercontent.com/13919702/136241502-eab319b4-b348-4b2d-b668-b4e68ecf3d30.jpg)

**Comment by ReinerBforartists** [comment]
Mesh nodes

![mesh](https://user-images.githubusercontent.com/13919702/136247132-9ce2ebdf-7afa-429d-b6e4-f4a4fae7f4df.jpg)
